enricher/fabric8/src/main/java/io/fabric8/maven/enricher/fabric8/ServiceDiscoveryEnricher.java

```diff
@@ -107,22 +118,42 @@ public ServiceDiscoveryEnricher(EnricherContext buildContext) {
 
         if (kind == Kind.SERVICE) {
 
-            tryCamelDSLProject();
+            if (!tried) {
+                tryCamelDSLProject();
+                //try more project types in future, last match wins
+                tried = true;
+            }
 
             if (discoverable != null) {
                 Map<String, String> labels = new HashMap<>();
-                String labelName = PREFIX + DISCOVERABLE;
+                String labelName = PREFIX;
                 labels.put(labelName, getConfig(Config.discoverable, discoverable));
-                if (log.isVerboseEnabled()) {
-                    log.verbose("Add %s label: %s=%s", PREFIX, 
-                            labelName, labels.get(labelName));
-                }
+                log.info("Add %s label: \"%s\" : \"%s\"", PREFIX, 
+                        labelName, labels.get(labelName));
                 return labels;
             }
         }
         return super.getLabels(kind);
     }
 
+    @Override
+    public void addMissingResources(final KubernetesListBuilder listBuilder) {
+        listBuilder.accept(new TypedVisitor<ServiceBuilder>() {
+
+            @Override
+            public void visit(ServiceBuilder serviceBuilder) {
+                
+                List<ServicePort> ports = serviceBuilder.buildSpec().getPorts();
+                if (! ports.isEmpty()) {
+                    ServicePort firstServicePort = ports.iterator().next();
+                    port = firstServicePort.getPort().toString();
```

Comment:
Shouldn't a user configured port always win ? That way a user can override the auto detection which couldn't be done otherwise. I would set port here only when port is not set. So, I also wouldn't allow a default value of '80' for the port but leave it empty. Only if no port is detected you might consider to default it to 80. But tbh, I would throw an exception then as its very likely the wrong port.

Reply:
@rhuss yes it should, so in line 101 we do: 
... getConfig(Config.port, port))
So the Config.port wins if specified.

3scale wants port to be 'required' value. So I have to specify it..

True that if no port is found an exception or warning is appropriate. I will add that.

Reply:
@KurtStam so if port is a required value when the even examine the port mapping of the service as the required value will allways win ? (i would remove the 80 as default value and throw an exception if getConfig(port) is giving null.

But you are right, the config in your code override the default port, **but** are you sure that `addMissingResource()` is called *before* `getAnnotations()`. The order is not defined, so as mentioned, carrying over state via member variables is dangerous. Why not skipping `getAnnotations()` and just do everything in the visitor atomically.

Well, but as I would remove the autodecection anyway, this is probably not a big deal anymore.

Reply:
Sure I can move everything into the addMissingResource, that makes sense. I do think we should prevent the user from forcing to specify a port as most services will only have 1 port.